### PR TITLE
DISABLE_SIGHTS aiming fix

### LINKED ISSUE #22043: Homemade Suppressor's DISABLE_SIGHTS flag prevents aiming
**Relation of #22031**

I have discovered the issue with the homemade suppressor's inability to steady aim and improve hit chance, however, I am uncertain of a proper implementation for a fix.
##  Effects of DISABLE_SIGHTS flag 
If there are no sights attached along with the homemade suppressor, the weapon's sight dispersion is set to MAX_RECOIL, 3000.
https://github.com/CleverRaven/Cataclysm-DDA/blob/3957dc7fab7e8f611df4c85a762d83efb9a790a0/src/item.cpp#L3997-L4005
Additionally, the weapon is set to have no sight speed modifier and no limit
https://github.com/CleverRaven/Cataclysm-DDA/blob/fdcc2da539577e68887153ff62a96050b92c6f93/src/character.cpp#L194-L196

## Results
Because of sight speed modifier being set to INT_MIN, aiming does not take place
https://github.com/CleverRaven/Cataclysm-DDA/blob/fdcc2da539577e68887153ff62a96050b92c6f93/src/character.cpp#L268-L271

Even if a sight speed modifier and a limit were set, aiming would still be pointless as the weapon's base dispersion is already at MAX_RECOIL.

It is quite late and I just can't seem to get a grip on a solution. I hope that the above evidence is correct and will assist in finding a solution.

### PR DESCRIPTION
**Pull request by TheFreckle**
**fixes #22043**
**relation of #22031**

Fixes the inability to aim if you have a homemade suppressor attached on a weapon, but do not have any sight modifications attached.

the values _sight_speed_modifier_ and _type->gun->sight_dispersion * 2_ are both arbitrary. I would appreciate some balance discussion on what values we want to use.

Tested with following parameters: As in **#22031** character is equipped with AR-15 with a 10 in all skills and attributes. Homemade Suppressor is then attached. These are the results of the item info screen

AR-15 no mods

AR-15 homemade suppressor

### CODE REVIEW

**Comment by kevingranade** [comment]
We need a value that represents how well a player can aim with no sights at all, it'll be a constant.
I'm not sure offhand what it should be, let's say.... 500?

**Comment by Coolthulhu** [comment]
MAX_RECOIL would be the perfect value for it. After all, this is the only sensible meaning it could have.
It could also be a good excuse to tie it to perception or even better - make perception actually matter when aiming in any context other than rifle scope+precise aim.

**Comment by kevingranade** [comment]
No, max_recoil is what you get when you point a gun in the general direction of a target, taking literally zero time to do so.  This value would represent the best you can do aiming a gun without the use of a sight, still sighting down the barrel, but with no markings or aligned features to help you.

**Comment by TheFreckle** [comment]
The value being set to MAX_RECOIL is the reason for this PR. You are unable to steady the weapon (as its base sight dispersion will be above MAX_RECOIL in most cases) and the chance to hit  at point blank range is extremely low (if I remember correctly, there was a 5% chance for a hit and a 10% chance for a grazing hit with a H&K smg). 

While I do believe perception could play a more important role, that change should be reserved for a future PR, as this PR only changes weapons who have the DISABLE_SIGHTS  flag and only have base sights to aim with (caused by attaching the homemade suppressor to a weapon).

I like the idea of an aiming constant, which is the value that could be used in any circumstance, barring any modifiers such as recoil or base sights. It could even be effected by perception. If I have time I will try to test some values tonight.

File: src/character.cpp [hunk 1]
```diff
@@ -196,6 +196,9 @@ std::pair<int, int> Character::get_best_sight( const item &gun, double recoil )
     if( !gun.has_flag( "DISABLE_SIGHTS" ) && effective_dispersion( gun.type->gun->sight_dispersion ) < recoil ) {
         sight_speed_modifier = 6;
         limit = effective_dispersion( gun.type->gun->sight_dispersion );
+    } else if (gun.has_flag("DISABLE_SIGHTS") && effective_dispersion(gun.type->gun->sight_dispersion) < recoil) {
```
Comment (Coolthulhu) [inline src/character.cpp]: Styling

**Comment by kevingranade** [comment]
Not sure why, but this messes up the aim UI.
If you reproduce #22043 and then aim as much as you can, your chance to hit reported by the UI is higher than your chance to hit with a precice aim shot, which should be impossible.

**Pull request merged**